src/api/procedures/setIssuancesData.ts

```diff
@@ -0,0 +1,86 @@
+import BigNumber from 'bignumber.js';
+
+import { SecurityToken } from '~/api/entities';
+import { PolymeshError, Procedure } from '~/base';
+import { ErrorCode, IssuanceData, Role, RoleType } from '~/types';
+import { numberToBalance, stringToIdentityId, stringToTicker } from '~/utils';
+
+export interface SetIssuancesDataParams {
+  issuances: IssuanceData[];
+}
+export type Params = SetIssuancesDataParams & {
+  ticker: string;
+};
+
+/**
+ * @hidden
+ */
+export async function prepareSetIssuancesData(
+  this: Procedure<Params, SecurityToken>,
+  args: Params
+): Promise<SecurityToken> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: { asset },
+      },
+    },
+    context,
+  } = this;
+  const { ticker, issuances } = args;
+
+  const securityToken = new SecurityToken({ ticker }, context);
+
+  const { isDivisible, totalSupply } = await securityToken.details();
+  const values = issuances.map(issuance => new BigNumber(issuance.balance));
+
+  values.forEach(value => {
+    if (isDivisible) {
+      if (value.decimalPlaces() > 6) {
+        throw new PolymeshError({
+          code: ErrorCode.ValidationError,
+          message: 'At most one balance exceeds the six decimals limit',
+        });
+      }
+    } else {
+      if (value.decimalPlaces()) {
+        throw new PolymeshError({
+          code: ErrorCode.ValidationError,
+          message: 'At most one balance has decimals',
+        });
+      }
+    }
+  });
+
+  const mintSupply = values.reduce((acc, next) => {
+    return acc.plus(next);
+  }, new BigNumber(0));
+
+  const limitTotalSupply = new BigNumber(Math.pow(10, 12));
+
+  if (totalSupply.plus(mintSupply).isGreaterThan(limitTotalSupply)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: `The total supply for "${ticker}" cannot be bigger than ${limitTotalSupply.toString()}`,
```

Comment:
This issuance operation will cause the total supply of "${ticker}" to exceed the maximum allowed (${MAX_TOKEN_AMOUNT})

Instead of using `toString`, use `toFormat` (tweak the parameters so that the number is printed with separators)